src/Http/Request.php

```diff
@@ -465,6 +465,9 @@ public function clearApplicationData() {
    * {@inheritdoc}
    */
   public function getApplicationData($key) {
+    if (!isset($this->applicationData[$key])) {
+      return NULL;
```

Comment:
It's a generic key value store in the form of an array. It will be empty if it was not set before. The `isset` is to avoid _undefined_ warnings.